usaspending_api/common/helpers/sql_helpers.py

```diff
@@ -246,6 +217,60 @@ def fetchall_to_ordered_dictionary(cursor):
     return [OrderedDict(zip(columns, row)) for row in cursor.fetchall()]
 
 
+def rowcount_fetcher(cursor):
+    """ Return the first value in the first row of the cursor. """
+    return cursor.rowcount
+
+
+def single_value_fetcher(cursor):
+    """ Return the first value in the first row of the cursor. """
+    return cursor.fetchall()[0][0]
+
+
+def execute_sql(sql, model=Award, fetcher=default_fetcher, read_only=True):
+    """
+    Executes a sql query against a database.
+
+    sql         - Can be either a sql string statement or a psycopg2 Composable
+                  object (Identifier, Literal, SQL, etc).
+    model       - A Django model that represents a database table germaine to your
+                  query.  If one is not supplied, Award will be used since it is
+                  fairly central to the database as a whole.
+    fetcher     - A function that accepts a live, post-execute cursor and returns
+                  whatever it wants.  This determines the return results of the
+                  function as a whole.
+    read_only   - Supply false here if you intend to perform updates against the
+                  database.
+
+    Returns query results in the format dictated by the fetcher.
+    """
+    connection = get_connection(model, read_only=read_only)
+    with connection.cursor() as cursor:
+        # Because django-debug-toolbar does not understand Composable queries,
+        # we need to convert the query to a string before executing it.
+        cursor.execute(convert_composable_query_to_string(sql, cursor=cursor))
+        return fetcher(cursor)
+
+
+def execute_update_sql(sql, model=Award):
+    """
+    Convenience function to execute a sql query against a database that performs
+    some sort of update (INSERT, UPDATE, DELETE, etc).  Returns the number of
```

Comment:
This is deceptive to less experienced sql devs such as myself. A function called `execute_modification_sql` or something of that ilk would make it much clearer that this isn't just for updates.

Reply:
Oh.  I see you addressed my previous question here.  So `execute_CUD_sql` is out then?

Reply:
After further reflection, I'm not sold on `modification`.  My first thought is that we're modifying the structure of the table.  In my head, `update` is a superset of `INSERT`, `UPDATE`, and `DELETE` since it is updating the data in the table.

Reply:
I'm also at a loss for the proper term then. Still bothered by the fact that `update` shadows the SQL keyword.